### DISCUSSION TITLE
Editing UI

### DISCUSSION THREAD

**Discussion by hiensumi** [cpeditor/cpeditor#1283] [Q&A]
Hello.

How to edit the UI in Qt Creator?
I saw people modifying it by using stylesheets but CP editor's source does not use any stylesheets. 
I would like to change the editor background color, button shapes...

**Comment by ouuan** [comment]
You can use the `setStyleSheet` Qt function.

**Reply by hiensumi** [reply]
@ouuan 
btw Im building on windows. I fixed the installation with third_party modules.
But the same error with the logs look like above appears when I start building cpeditor

**Reply by ouuan** [reply]
You may refer to https://github.com/cpeditor/cpeditor/blob/master/.github/workflows/release.yml. Or you can build a version without KSH and ECM, like v6.11.

**Reply by hiensumi** [reply]
@ouuan Please help me.
I can not figure out how to change stylesheet of QMainWindow or other widgets in other classes.
![image](https://github.com/cpeditor/cpeditor/assets/47080203/21833579-e833-4892-9a22-7eb5ad257892)

**Comment by ouuan** [comment]
I'm not sure what you want to do. If you want to know how to write the stylesheets, please read the Qt documentation, because I don't know, either. If you want to know where to call `setStyleSheet`, well, you may call it in the constructors of the corresponding classes.

**Reply by hiensumi** [reply]
Im trying to sync the editor theme with UI theme like other IDE :)))
![image](https://github.com/cpeditor/cpeditor/assets/47080203/de028245-f20a-4375-b251-4b69c0f70d37)

**Reply by hiensumi** [reply]
@ouuan how to pack cpeditor src to an installer for every os?

**Comment by ouuan** [comment]
> how to pack cpeditor src to an installer for every os?

Refer to https://github.com/cpeditor/cpeditor/blob/master/.github/workflows/release.yml

BTW, please don't always reply under the same comment when the topic is changed. You may start a new thread or at least a new comment.

**Reply by hiensumi** [reply]
> > how to pack cpeditor src to an installer for every os?
> 
> Refer to https://github.com/cpeditor/cpeditor/blob/master/.github/workflows/release.yml
> 
> BTW, please don't always reply under the same comment when the topic is changed. You may start a new thread or at least a new comment.

I want to pack for every OS on my current OS. :))
The workflow just displays packing for the current one. :>

**Reply by ouuan** [reply]
You can run it via GitHub Actions, e.g. in your fork

**Comment by hiensumi** [comment]
Help please!
I can not figure out how to identify which theme is currently being used in CP editor .

**Reply by hiensumi** [reply]
I am trying to get KSyntaxHighlighting current theme is being used.

**Reply by ouuan** [reply]
You have found the `CodeEditor::setTheme` function, so just search for where it is called to see how the current theme is fetched.
